internal/vault/cache_storage.go

```diff
@@ -585,8 +590,9 @@ func (c *defaultClientCacheStorage) incrementOperationCounter(operation string,
 type ClientCacheStorageConfig struct {
 	// EnforceEncryption for persisting Clients i.e. the controller must have VaultTransitRef
 	// configured before it will persist the Client to storage. This option requires Persist to be true.
-	EnforceEncryption bool
-	HMACSecretObjKey  ctrlclient.ObjectKey
+	EnforceEncryption           bool
+	PruneVaultTokensOnUninstall bool
```

Comment:
I fell like this variable name could be confusing, what does uninstall mean for ClientCacheStorage?. I wonder if it should be something related to how the secrets are created - it could even just be pointer to some `OwnerReferences`, so if those are set just include the owner refs for any stored Secret?

Reply:
I agree. I will move the logic to populate `OwnerReferences` during the setup and drop other fields.